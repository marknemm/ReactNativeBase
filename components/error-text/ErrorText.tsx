import { Text } from '@rneui/themed';
import type { ErrorTextProps } from './ErrorText.interfaces';
import { useStyles } from './ErrorText.styles';

/**
 * The {@link ErrorText} component.
 *
 * @param props The component {@link ErrorTextProps}.
 * @returns The {@link ErrorText} component.
 */
const ErrorText: React.FC<ErrorTextProps> = ({
  children,
  error,
  style,
  ...textProps
}) => {
  const styles = useStyles({ style });

  return (children || error) && (
    <Text
      style={styles.style}
      {...textProps}
    >
      {
        children || ((error instanceof Error)
          ? error.message
          : error)
      }
    </Text>
  );
};

export type * from './ErrorText.interfaces';
export default ErrorText;
