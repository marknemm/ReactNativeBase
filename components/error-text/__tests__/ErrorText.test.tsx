import ErrorText from '@components/error-text/ErrorText';
import { Text } from '@rneui/themed';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen } from '@testing-library/react-native';

describe('<ErrorText />', () => {
  describe('displays error', () => {
    it('displays error message', () => {
      const errorMsg = 'Test error message';
      render(
        <ErrorText error={errorMsg} />,
        { wrapper: AppProvider }
      );

      const errorText = screen.getByText(errorMsg);
      expect(errorText).toBeDefined();
    });

    it('displays error message from Error object', () => {
      const error = new Error('Test error message');
      render(
        <ErrorText error={error} />,
        { wrapper: AppProvider }
      );

      const errorText = screen.getByText(error.message);
      expect(errorText).toBeDefined();
    });

    it('displays custom children error message', () => {
      const errorMsg = 'Test error message';
      render(
        <ErrorText>
          <Text>{errorMsg}</Text>
        </ErrorText>,
        { wrapper: AppProvider }
      );

      const errorText = screen.getByText(errorMsg);
      expect(errorText).toBeDefined();
    });
  });

  describe('custom styles', () => {
    it('applies custom styles', () => {
      const errorMsg = 'Test error message';
      const style = { color: 'red' };
      render(
        <ErrorText
          error={errorMsg}
          style={style}
        />,
        { wrapper: AppProvider }
      );

      const errorText = screen.getByText(errorMsg);
      expect(errorText).toHaveStyle(style);
    });
  });

  describe('snapshots', () => {
    it('renders correctly with no error or children', () => {
      render(
        <ErrorText />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with error message', () => {
      render(
        <ErrorText error="Test error message" />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
