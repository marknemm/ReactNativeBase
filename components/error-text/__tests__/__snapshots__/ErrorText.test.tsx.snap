// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorText /> snapshots renders correctly with error message 1`] = `
<Text
  accessibilityRole="text"
  style={
    {
      "color": "#ff3b30",
      "fontFamily": undefined,
      "fontSize": 23.57142857142857,
      "fontStyle": "normal",
      "fontWeight": "normal",
      "marginBottom": 0,
      "paddingHorizontal": 6.285714285714286,
      "textAlign": "left",
      "textDecorationLine": "none",
      "textDecorationStyle": "solid",
    }
  }
>
  Test error message
</Text>
`;

exports[`<ErrorText /> snapshots renders correctly with no error or children 1`] = `null`;
